enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,264 @@
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# scorecard-storage-option
+
+This enhancement provides scorecard users a means to persist reports or other output from their scorecard tests to a persistent location that can be collected
+by scorecard.
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+None
+
+## Summary
+
+QE and other test writers sometimes require storage within their test environment (e.g. test container) to write reports and other test output material.  Today within scorecard this is not possible since a persistent volume is not available within the test container.  As the test completes, the container ends, leaving only the container log on the k8s cluster until the Pod is deleted.
+
+This enhancement would offer an optional mechanism whereby test writers have a persistent volume mounted within their test containers, provisioned, enabled, and cleaned up by scorecard.
+
+
+## Motivation
+
+Allowing scorecard tests a place to persist test output is key to running advanced tests and supporting complex test output.
+
+QE teams for example might already have tests which produce test output files, in order for them to integrate their tests into scorecard, they need a persistent volume to write test results into.
+
+This feature enables a wider adoption of tests that can be executed within scorecard.
+
+### Goals
+
+Goals of this proposal include:
+* provide a persistent volume for scorecard test developers
+* Collect persistent volume output when scorecard completes
+
+### Non-Goals
+
+This enhancement doesn't change scorecard's test result format (v1alpha3) so this is not a breaking change.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+A scorecard user can specify for a given test that they want persistent storage to be made available to that test.  This is specified within the scorecard configuration file as a label.  For example, a user would specify they want storage for a test by specifying
+in the scorecard configuration file the `storage` label as in the following example:
+
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+      storage: true
+```
+
+#### Story 2
+A scorecard user can optionally specify storage class details within the scorecard configuration file to override storage defaults. Volume size, access mode, and storage class names can be specified as follows:
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    storage: 
+      provision: true
+      size: 1Gi
+      class: standard
```

Comment:
can we just inline the podtemplate volume spec?

Reply:
scorecard's config.yaml was designed to be user facing, so its pretty simple in design on purpose where QE people can just fill in the values they need in order to run their tests.  But I can take a look at what the pod volume spec would look like here instead of the labels.

Reply:
I updated this enhancement to use PersistentVolume specs instead of labels within the config.yaml file.